incubator/hnc/api/v1alpha2/hierarchy_types.go

```diff
@@ -0,0 +1,298 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"fmt"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// Constants for types and well-known names
+const (
+	Singleton               = "hierarchy"
+	HierarchyConfigurations = "hierarchyconfigurations"
+)
+
+// Constants for labels and annotations
+const (
+	MetaGroup                = "hnc.x-k8s.io"
+	LabelInheritedFrom       = MetaGroup + "/inheritedFrom"
+	FinalizerHasSubnamespace = MetaGroup + "/hasSubnamespace"
+	LabelTreeDepthSuffix     = ".tree." + MetaGroup + "/depth"
+	AnnotationManagedBy      = MetaGroup + "/managedBy"
+)
+
+// Condition codes. *All* codes must also be documented in the comment to Condition.Code, be
+// inserted into AllCodes, and must have an entry in ClearConditionCriteria, set in init() in this
+// file.
+//
+// Please keep this list in alphabetic order.
+const (
+	CannotPropagate           Code = "CannotPropagateObject"
+	CannotUpdate              Code = "CannotUpdateObject"
+	CritAncestor              Code = "CritAncestor"
+	CritCycle                 Code = "CritCycle"
+	CritDeletingCRD           Code = "CritDeletingCRD"
+	CritParentMissing         Code = "CritParentMissing"
+	SubnamespaceAnchorMissing Code = "SubnamespaceAnchorMissing"
+)
+
+var AllCodes = []Code{
+	CannotPropagate,
+	CannotUpdate,
+	CritAncestor,
+	CritCycle,
+	CritDeletingCRD,
+	CritParentMissing,
+	SubnamespaceAnchorMissing,
+}
+
+// ClearConditionCriterion describes when a condition should be automatically cleared based on
+// forest changes. See individual constants for better documentation.
+type ClearConditionCriterion int
+
+const (
+	CCCUnknown ClearConditionCriterion = iota
+
+	// CCCManual indicates that the condition should never be cleared automatically, based on the
+	// structure of the forest. Instead, the reconciler that sets the condition is responsible for
+	// clearing it as well.
+	CCCManual
+
+	// CCCAncestor indicates that the condition should always exist in the namespace's ancestors, and
+	// should be cleared if this is no longer true.
+	CCCAncestor
+
+	// CCCSubtree indicates that the condition should always exist in the namespace's subtree (that
+	// is, the namespace itself or any of its descendants), and should be cleared if this is no longer
+	// true.
+	CCCSubtree
+)
+
+// ClearConditionCriteria is initialized in init(). See ClearConditionCriterion for more details.
+var ClearConditionCriteria map[Code]ClearConditionCriterion
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +kubebuilder:object:root=true
+// +kubebuilder:skipversion
```

Comment:
This is the `skipversion` marker added to the v1alpha2 copy from v1alpha1. The `unservedversion` seemed not working, see [issue](https://github.com/kubernetes-sigs/controller-tools/issues/462).